### PR TITLE
Fix some typo, some clippy warnings and some impl order find by intel…

### PR DESCRIPTION
**Pull request by remilauzier**
…lij idea

### CODE REVIEW

File: book/src/entities-components.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Entities and their components
 
-The core of ECS is knowing that an entity is a collection of components. These
+The core of ECS knows that an entity is a collection of components. These
```
Comment (cdsupina) [inline book/src/entities-components.md:3]: Perhaps we should change this phrasing to:

"The key to understanding ECS is knowing that..."

File: thetawave_game/src/states/paused.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ impl SimpleState for PausedState {
         event: StateEvent,
     ) -> SimpleTrans {
         if let StateEvent::Window(event) = &event {
-            if is_key_down(&event, VirtualKeyCode::Escape) {
+            if is_key_down(event, VirtualKeyCode::Escape) {
```
Comment (cdsupina) [inline thetawave_game/src/states/paused.rs:15]: Where are you getting these lints from? I'm using Clippy as well and it isn't identifying this one.

File: thetawave_lib/src/audio/audio_utils.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ pub fn initialize_audio(world: &mut World) {
         for (sound_name, sound_file) in sound_data.iter() {
             sound_effects.insert(
                 sound_name.to_owned(),
-                load_audio_track(&loader, &world, &*("audio/".to_string() + sound_file)),
+                load_audio_track(&loader, world, &*("audio/".to_string() + sound_file)),
```
Comment (cdsupina) [inline thetawave_lib/src/audio/audio_utils.rs:31]: Same question as above.

File: thetawave_lib/src/spawn/components/spawn.rs [hunk 4]
```diff
@@ -61,43 +61,43 @@ impl AutoSpawnerComponent {
             match &self.child_entity_type {
                 SpawnableType::Mob(mob_type) => {
                     mobs_resource.spawn_mob(
-                        &mob_type,
+                        mob_type,
```
Comment (cdsupina) [inline thetawave_lib/src/spawn/components/spawn.rs:64]: Were these also suggested by Clippy?

**Review by cdsupina — COMMENTED** [review]
Looks good, I only wanted to know where you are getting these lints from. I'm using clippy as well and I don't see any of lints you addressed indicated.

File: thetawave_game/src/states/paused.rs (hunk 2 above)
Comment (remilauzier) [reply thetawave_game/src/states/paused.rs:15]: Clippy nightly. Needless borrow and needless reference.

Comment (remilauzier) [reply thetawave_game/src/states/paused.rs:15]: https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

Comment (cdsupina) [reply thetawave_game/src/states/paused.rs:15]: I see, thanks for the clarification.

File: book/src/entities-components.md (hunk 1 above)
Comment (cdsupina) [reply book/src/entities-components.md]: Just fix this phrasing and I'll merge it.

**Review by cdsupina — APPROVED** [review]
Good work! And thanks for the clarification about Clippy!

**Pull request merged**